src/csharp/Microsoft.Spark/Network/DefaultSocketWrapper.cs

```diff
@@ -6,6 +6,7 @@
 using System.IO;
 using System.Net;
 using System.Net.Sockets;
+using Microsoft.Spark.Interop;
 using Microsoft.Spark.Services;
```

Comment:
Don't you need to update `JvmBridge`? Btw, how is this scenario tested?

Reply:
How about `DataFrame.GetRows`?

Reply:
I am passing the IP address through `connect` and binding to `IPAddress.Any` but I am getting the following error:
`[Exception] [JvmBridge] No connection could be made because the target machine actively refused it 40.125.65.29:5567`
`40.125.65.29` is my VM where dotnet backend is running and I have added appropriate (I think) inbound and outbound rules for the 5567 port. Not sure what I am missing.

Reply:
shouldn't the jvm side be `channelFuture = bootstrap.bind(new InetSocketAddress("0.0.0.0", portNumber))`? btw, I think this could cause a security concern.

Reply:
I think perhaps we need to differentiate between two use cases. I would agree that if running the backend in a production scenario, we wouldn't want to allow connections from any client. We should be secure by default. However, when running in debug more, the typical use case is that it's either running in a container, or in our case, just a WSL2 Ubuntu instance. To make it easier to use and test, it's nice to be able to run tests and code in Windows and connect to the DotNetRunner in Linux/Spark/WSL2 using localhost. This has been supported since Windows Build 18945:
https://docs.microsoft.com/en-us/windows/wsl/networking#accessing-linux-networking-apps-from-windows-localhost
However, it only works if the server binds to 0.0.0.0. This bug tracked the problem but I think the closure was premature:
https://github.com/microsoft/WSL/issues/4338

If I run in my WSL2 environment sudo nc -l 5567 I can connect to that port from Windows in Edge using localhost:5567, so I know the port works. The difference as I see it is that nc binds to 0.0.0.0 and not localhost. Since the dotnet spark solution binds to localhost, things do not work. I'd like to test a version that allows 0.0.0.0 (by override/command line) and see if we can get this PR over the finish line.